haskell-overlays/android/default.nix

```diff
@@ -26,4 +26,10 @@ self: super: {
     dontStrip = true;
     enableSharedExecutables = false;
   });
+
+  # HACK(matthewbauer):
+  # Temporary fix for https://github.com/ekmett/free/issues/176
+  # Optimizations are broken on iOS for some modules.
```

Comment:
What's the relationship between iOS (as the comment says here) and Android (the file we're in)?  Also, what needs to happen for us to be able to pull out these hacks?  Let's make issues in the appropriate places.